python/lsst/ip/diffim/subtractImages.py

```diff
@@ -162,8 +162,10 @@ def setDefaults(self):
         self.makeKernel.kernel.active.spatialBgOrder = 2
 
     def validate(self):
-        if self.forceCompatibility:
-            self.mode = "convolveTemplate"
+        if self.forceCompatibility and not (self.mode == "convolveTemplate"):
+            msg = "forceCompatibility=True requires mode='convolveTemplate'"
```

Comment:
Oh, I forgot to include one thing in my suggestion: ` but mode was '{self.mode}'"` at the end of this message (and make it an f-string).